### ISSUE TITLE
Audio List - specific play song

### ISSUE THREAD

**Issue by Iulxim** [shinaBR2/shinabr2-world#33]
Hi.

Is there a way to play a specific song from an audioList?
From the API exposed all i can find is onPlay which is playing ref.current song.

Thank you!

**Comment by shinaBR2** [comment]
Hi @Iulxim 

Here is an example of code for that purpose: https://github.com/shinaBR2/shinabr2-world/blob/main/packages/ui/src/listen/minimalism/music-widget/MusicWidget.tsx

Basically, the core logic here is I use the index (number) in the application state.

**Comment by Iulxim** [comment]
Thank you !

**Comment by shinaBR2** [comment]
Hi @Iulxim there is an example with the `getControlsProps` here: https://www.npmjs.com/package/@shinabr2/life-core
Could you try it out?

**Comment by Iulxim** [comment]
Hey. Yes, based on this example is what i have been playing around with. But `getControlsProps` only exposes 
`onPlay`, `onPrev`, `onNext`, `onShuffle`, `onChangeLoopMode`. 

I have created a [sandbox](https://codesandbox.io/s/dreamy-mountain-2e8cfx?file=/src/App.js)

My intention is when i select a song from a list to play, if i hit that button again, it should pause, if i select another it should play, which i managed to do based on your previous help, but i cant treat the pause state with onPlay method or at least i cant figure it out.

**Comment by shinaBR2** [comment]
Ah, I see your intention @Iulxim that's not my intention because I want the separation of concern.
Let's say we have two actions here: select an audio item, and control (play/pause).
Your intention is to combine those actions into one single click, you can do it outside my library because it will make the hook easier to reusable.

So in your example, in your `onSelect` function you need to add some more additional logic:
- If the selected audio item is the same as the current playing audio, pause it.
- Otherwise, play.
Basically, you need to handle your custom logic and this kind of logic should not be included in the `useSAudioPlayer`. Hope this makes sense. Feel free to ask for help.
